analyzer/src/relations.rs

```diff
@@ -46,85 +62,86 @@ pub struct ResolvedSymbol {
+    pub object_id: LocalId,
 }
 
 impl ResolvedSymbol {
-    pub fn new(source: SourceObjectId, object_id: ObjectId) -> Self {
+    pub fn new(source: SourceId, object_id: LocalId) -> Self {
         Self { source, object_id }
     }
 }
 
-/// The state of an object
+/// The state of a relation
 ///
-/// The [SymbolResolver] only handles objects marked as [ObjectState::Unresolved]
-/// and will attempt to resolve it by updating his state to [ObjectState::Resolved]
-/// If the resolution fails, for any reason, the object is marked as dead ([ObjectState::Dead])
-/// which should imply a diagnostic.
-/// This state prevents the resolver to attempt to resolve again unresolvable symbols on next cycles.
+/// The [SymbolResolver] only attempts to resolve relations marked as [RelationState::Unresolved]
+/// If the resolution fails, for any reason, the object is marked as dead ([RelationState::Dead])
+/// which in most case implies a diagnostic.
+/// The dead state prevents the resolver to attempt to resolve again unresolvable symbols on next cycles.
+/// If the relation was successfully resolved, the state is then [RelationState::Resolved], containing the
+/// resolved symbol and targeted environment.
 #[derive(Debug, Clone, Copy, Hash, PartialEq)]
-pub enum ObjectState {
+pub enum RelationState {
     Resolved(ResolvedSymbol),
     Unresolved,
     Dead,
 }
 
-impl ObjectState {
+impl RelationState {
     pub fn expect_resolved(self, msg: &str) -> ResolvedSymbol {
         match self {
-            ObjectState::Resolved(resolved) => resolved,
+            RelationState::Resolved(resolved) => resolved,
             _ => panic!("{}", msg),
         }
     }
 }
 
 #[derive(Debug, Clone, Hash, PartialEq)]
-pub struct Object {
+pub struct Relation {
     /// The environment's id that requested this object resolution.
-    pub origin: SourceObjectId,
+    pub origin: SourceId,
 
-    /// This object's state.
-    /// See [ObjectState] for more details
-    pub state: ObjectState,
+    /// This relation's state.
+    /// See [RelationState] for more details
+    pub state: RelationState,
 }
 
-impl Object {
-    pub fn unresolved(origin: SourceObjectId) -> Self {
+impl Relation {
+    pub fn unresolved(origin: SourceId) -> Self {
         Self {
             origin,
-            state: ObjectState::Unresolved,
+            state: RelationState::Unresolved,
         }
     }
 
-    pub fn resolved(origin: SourceObjectId, resolved: ResolvedSymbol) -> Self {
+    pub fn resolved(origin: SourceId, resolved: ResolvedSymbol) -> Self {
         Self {
             origin,
-            state: ObjectState::Resolved(resolved),
+            state: RelationState::Resolved(resolved),
         }
     }
 }
 
 /// A collection of objects that are tracked globally and may link to each other.
 #[derive(Debug, Default)]
 pub struct Relations {
-    /// The objects that need resolution that are tracked globally.
+    /// The tracked relations.
```

Comment:
```suggestion
    /// The globally tracked relations.
```

Reply:
idk, there's no "local" relations so, maybe `tracked relations between environments` could be an option ?

Reply:
Local relations exist as they just share the same local id. *between environments* translates great the purpose of this system of relations.